homeassistant/components/intellifire/__init__.py

```diff
@@ -60,79 +60,82 @@ def _construct_common_data(entry: ConfigEntry) -> IntelliFireCommonFireplaceData
     )
 
 
-async def _async_pseudo_migrate_entry(
-    hass: HomeAssistant, config_entry: ConfigEntry
-) -> ConfigEntry:
-    """Update configuration entry to latest VERSION 1 format."""
-    new = {**config_entry.data}
-    # Rename Host to IP Address
-    new[CONF_IP_ADDRESS] = new.pop("host")
-
-    username = config_entry.data[CONF_USERNAME]
-    password = config_entry.data[CONF_PASSWORD]
-
-    # Create a Cloud Interface
-    async with IntelliFireCloudInterface() as cloud_interface:
-        await cloud_interface.login_with_credentials(
-            username=username, password=password
-        )
-
-        # See if we can find the fireplace first by serial and then secondly by IP.
-
-        serial = config_entry.title.replace("Fireplace ", "")
-
-        # If serial matches the hex style pattern we'll assume its good
-        valid_serial = bool(re.match(r"^[0-9A-Fa-f]{32}$", serial))
-
-        new_data = (
-            cloud_interface.user_data.get_data_for_serial(serial)
-            if valid_serial
-            else None
-        )
-
-    if not new_data:
-        new_data = cloud_interface.user_data.get_data_for_ip(new[CONF_IP_ADDRESS])
-
-    if not new_data:
-        raise ConfigEntryAuthFailed
+async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Migrate entries."""
+    LOGGER.debug(
+        "Migrating configuration from version %s.%s",
+        config_entry.version,
+        config_entry.minor_version,
+    )
 
-    # Find the correct fireplace
-    if new_data is not None:
-        new[CONF_API_KEY] = new_data.api_key
-        new[CONF_WEB_CLIENT_ID] = new_data.web_client_id
-        new[CONF_AUTH_COOKIE] = new_data.auth_cookie
-
-        new[CONF_IP_ADDRESS] = new_data.ip_address
-        new[CONF_SERIAL] = new_data.serial
-
-        config_entry.version = 1
-        hass.config_entries.async_update_entry(
-            config_entry,
-            data=new,
-            options={CONF_READ_MODE: "local", CONF_CONTROL_MODE: "local"},
-            unique_id=serial,
-        )
-        LOGGER.debug("Pseudo Migration %s successful", config_entry.version)
+    if config_entry.version == 1:
+        new = {**config_entry.data}
+
+        if config_entry.minor_version < 2:
+            try:
+                new[CONF_IP_ADDRESS] = new.pop("host")
```

Comment:
We should not pop the host. The host is always there in the old schema right?

Reply:
I think I asked this before, why do we actually change from HOST to IP_ADDRESS? Any significance?

Reply:
It better aligns with lib.